### DISCUSSION TITLE
Cannot connect to local PostgreSQL DB

### DISCUSSION THREAD

**Discussion by mhdaouas** [Budibase/budibase#1839] [Help] [answered]
Hello,

I'm unable to connect to my PSQL DB. I tried the following IP addresses without success:

- 127.0.0.1
- localhost
- 0.0.0.0
- 192.168... (local IP)
- 172.17.0.1 (docker IP)

It seems like the docker container is unable to reach my local DB.

However I could connect to a local API from a Django server by serving it on my 192... IP.
I even tried to make PSQL listen on 0.0.0.0 and 192... without success.

Please help!

**Comment by shogunpurple** [comment]
Hi @mhdaouas - is your postgres running inside docker as well, or just on the machine?

**Reply by mhdaouas** [reply]
on my machine

**Reply by shogunpurple** [reply]
Thanks. It seems odd that your machines full local IP address does not work when entered correctly. Just to debug - does `host.docker.internal` work?

**Reply by mhdaouas** [reply]
How can I check that? Is it a service?

**Reply by shogunpurple** [reply]
I just mean replace the IP address you are using in your datasource (127.0.0.1 etc.) with `host.docker.internal`

EDIT: try `172.17.0.1` also. I'm trying to see if docker is resolving your IP address internally rather than on the host machine

**Reply by mhdaouas** [reply]
I already tried it without success.

**Reply by shogunpurple** [reply]
@mhdaouas ok thank you for debugging that. I’m not too sure what the problem could be in that case - I will try to recreate in my own environment with a Postgres DB

**Reply by Manishkparmar80** [reply]
I am also having same issue connecting to SQL Sever DB which is hosted on local machine. I tried Local host, 172.17.0.1 but no luck.  I can connect SQL DB which is hosted on other machine which is connected over network, but issue is only observed when we connect the DB installed on local machine on which docker is running.

**Comment by LivingWithHippos** [comment, marked as answer]
Docker has its own networks isolated from the local ones, you can either tell your local db to listen to the budibase network or make budibase run on the local network.  Run `docker inspect CONTAINER_NAME -f "{{json .NetworkSettings.Networks }}"`  to find out details about your budibase network. This is a docker "problem", not a budibase problem. See something like https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach

**Reply by shogunpurple** [reply]
Thanks @LivingWithHippos - I was unable to recreate the problem above when I used the `host.docker.internal` as well. I think the issues that people were having were docker environment/networking issues.

**Comment by Manishkparmar80** [comment]
@LivingWithHippos  Yes, by using server name as **host.docker.internal** instead of Localhost , MS SQL DB connected successfully with Budibase.   From many days, i was struggling with this issue and with your help it is now got resolved. Thanks.

**Comment by chiptech-mitchell** [comment]
My Budibase is on a server and to connect to my local docker instance running Postgres I had to open the Postgres port in the firewall and use my local IPv4 address as the host

**Comment by btongkm** [comment]
I am having same issue with docker in synology. I have 2 docker containers up and running, One is Budibase and another one is postres. My Budibase is link to postgres container when launch. When I set up my application in Budibase datasource as PostgresSOL, it require the host.  I tried 1) host.docker.internal and  2) local host both didn't work. Is there anyone know what I should enter into the host?

**Reply by JayP718** [reply]
Try your own IP address

**Reply by btongkm** [reply]
I finally got it works. Now I can add PostgresSQL as my datasouce by adding the PostgresSQL container IP in the host field. Make sure the database name, username and password are set correctly. I can add table either directly in Budibase app, or in pgAdmin.
